### PR TITLE
CCE cluster node cloud layer integration

### PR DESCRIPTION
**Pull request by tischrei**

### CODE REVIEW

File: otcextensions/sdk/cloud/cce.py [hunk 1]
```diff
@@ -199,3 +199,216 @@ def delete_cce_cluster(
...
+                wait_args['wait'] = wait_timeout
+
+            self.cce.wait_for_job(obj.job_id, **wait_args)
+            obj = self.cce.get_cluster(obj.id)
+
+        return obj
+
+    def delete_cce_cluster_node(
+        self,
+        wait=True, wait_timeout=180, wait_interval=5,
```
Comment (lego963) [inline otcextensions/sdk/cloud/cce.py:383]: I suppose, that 3mins in `wait_timeout` is very little time

**Review by lego963 — CHANGES_REQUESTED** [review]

File: otcextensions/sdk/cce/v3/cluster_node.py [hunk 2]
```diff
@@ -47,6 +47,43 @@ class PublicIPSpec(resource.Resource):
     floating_ip = resource.Body('eip', type=dict)
```
Comment (anton-sidelnikov) [inline otcextensions/sdk/cce/v3/cluster_node.py:47]: pls look at 41-46, may be we can uncomment bandwidth?

File: examples/cce/find_cluster_node.py [hunk 3]
```diff
@@ -19,8 +19,8 @@
 conn = openstack.connect(cloud='otc')
 
 
-cluster = "cluster_name_or_id"
+cluster = 'name_or_id'
 cluster = conn.cce.find_cluster(cluster)
-node = "node_name_or_id"
+node = 'name_or_id'
```
Comment (anton-sidelnikov) [inline examples/cce/find_cluster_node.py]: you can insert name right into find_cluster_node

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: otcextensions/sdk/cloud/cce.py [hunk 4]
```diff
@@ -199,3 +199,216 @@ def delete_cce_cluster(
...
+        :param str cluster: CCE cluster attached to.
+        :param list data_volumes: List of Data volumes attached to the
+            cluster node.
+        :param str dedicated_host: ID of the Dedicated Host to which
+            nodes will be scheduled.
+        :param str ecs_group: ID of the ECS group where the CCE node can
+            belong to.
+        :param str fault_domain: The node is created in the specified fault
+            domain.
+        :param str flavor: Flavor.
```
Comment (lego963) [inline otcextensions/sdk/cloud/cce.py]: can you specify `FlavorId` or `FlavorName`? (If it matters)

File: otcextensions/sdk/cloud/cce.py [hunk 5]
```diff
@@ -199,3 +199,216 @@ def delete_cce_cluster(
...
+                                     'between 100 and 32768 GB.')
+                spec['dataVolumes'].append({
+                    'volumetype': key.upper(),
+                    'size': item[key]
+                })
+
+        if availability_zone:
+            spec['az'] = availability_zone
+        if count:
+            spec['count'] = count
```
Comment (lego963) [inline otcextensions/sdk/cloud/cce.py]: you have already set in 303 line

File: otcextensions/sdk/cloud/cce.py (hunk 1 above)
Comment (tischrei) [reply otcextensions/sdk/cloud/cce.py:383]: It ran through every, so I did not suppose to increase the time right now. Deletion of the node is fast. Deletion of the cluster takes a lot of time.

Comment (tischrei) [reply otcextensions/sdk/cloud/cce.py:383]: increased to 300

File: otcextensions/sdk/cce/v3/cluster_node.py (hunk 2 above)
Comment (tischrei) [reply otcextensions/sdk/cce/v3/cluster_node.py:47]: Nope, external IP is something different. It is the management IP to use kubectl to manage the cluster resources. For ECS and in this case cce node the name floating_ip seems to be the common name for it: https://opendev.org/openstack/openstacksdk/src/branch/master/openstack/network/v2/floating_ip.py

Comment (tischrei) [reply otcextensions/sdk/cce/v3/cluster_node.py:47]: I can't find "bandwidth" anywhere in the document.

File: examples/cce/find_cluster_node.py (hunk 3 above)
Comment (tischrei) [reply examples/cce/find_cluster_node.py]: Yes I can, but I used this simpler form to distinguish in those example file, where parameters are coming from. Nevertheless, I changed it as requested.

File: otcextensions/sdk/cloud/cce.py (hunk 4 above)
Comment (tischrei) [reply otcextensions/sdk/cloud/cce.py]: done

File: otcextensions/sdk/cloud/cce.py (hunk 5 above)
Comment (tischrei) [reply otcextensions/sdk/cloud/cce.py]: oki

**Review by anton-sidelnikov — DISMISSED** [review]

File: examples/cce/create_cluster_node.py [hunk 6]
```diff
@@ -20,34 +20,49 @@
...
-        "flavor": "cce.s1.small",
-        "version": "v1.13.10-r0",
-        "az": "eu-de-01",
-        "supportIstio": True,
-        "hostNetwork": {
-            "vpc": "26ca2783-dc40-4e3a-95b1-5a0756441e12",
-            "subnet": "25d24fc8-d019-4a34-9fff-0a09fde6a9cb",
-            "SecurityGroup": "f9ae0767-25be-44fc-a21c-5b8a0da66dec"
+        "flavor": "s2.large.2",
+        "az": "eu-de-02",
```
Comment (lego963) [inline examples/cce/create_cluster_node.py:36]: We have problems with storage in az2, can you change to everywhere az1?

**Review by lego963 — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**